xdsl/dialects/riscv.py

```diff
@@ -159,6 +161,31 @@ class RISCVOp(Operation, ABC):
     pass
 
 
+class RISCVPrinterInterface(ABC):
+    """
+    This interface is used so that other dialects can extend RISC-V printing
+    without having to modify printing code or the risc-v dialect base.
+    """
+
+    @abstractmethod
+    def riscv_printed_name(self) -> str:
+        """
+        Give the name of the RISC-V instruction
+        """
+        raise NotImplementedError()
```

Comment:
I am always unsure what the state of the art (if you want to call Python art) is on abstract method bodies. Pass? Return? NotImplementedError?

Feel free to ignore this comment if you know better :)

Reply:
I like `raise NotImplemented`

Reply:
I would argue that NotImplementedError implies that it can get implemented.
I would prefer `...` or `assert False, "description"`, but no strong feelings

Reply:
I agree with @math-fehr